spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -173,12 +173,22 @@ protected void sendMessage(Message<?> message) {
 		catch (RuntimeException e) {
 			MessageChannel errorChannel = getErrorChannel();
 			if (errorChannel != null) {
-				this.messagingTemplate.send(errorChannel, new ErrorMessage(e));
+				this.messagingTemplate.send(errorChannel, buildErrorMessage(e));
```

Comment:
The question relates to the change here: https://github.com/spring-projects/spring-integration-kafka/pull/163/files#diff-b49c842da794eb3622242c8acb03e1c1R290

I don't know the story yet, but let @garyrussell to answer!